two-entity-poc/src/util/generate-error.ts

```diff
@@ -0,0 +1,5 @@
+export const getErrorObject = (errMsg: string, statusCode: number) => {
+    const error = new Error(errMsg);
+    (error as any).statusCode = statusCode;
+    return error;
+}
```

Comment:
if we have to create a newer version of error;

one good approach is 

class AppError extends Error {
  constructor(message, statusCode) {
    super(message);
    this.statusCode = statusCode;
    this.status = `${statusCode}`.startsWith('4') ? 'fail' : 'error';
    this.isOperational = true;

    Error.captureStackTrace(this, this.constructor);
  }
}

module.exports = AppError;

then you can create new AppError(message, statusCode)

Reply:
Need to explore this approach. Will do the same in next assignment.